src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs

```diff
@@ -31,83 +31,57 @@ public Task CreateQueueIfNecessary(QueueBindings queueBindings, string identity)
-                {
-                    Logger.Debug("Queue exists, going to set permissions.");
-                    SetPermissionsForQueue(queuePath, identity);
-                    return;
-                }
-
                 Logger.Warn("Queue " + queuePath + " does not exist.");
                 Logger.Debug("Going to create queue: " + queuePath);
 
-                CreateQueue(queuePath, identity, settings.UseTransactionalQueues);
-            }
-            catch (MessageQueueException ex)
-            {
-                if (msmqAddress.IsRemote && (ex.MessageQueueErrorCode == MessageQueueErrorCode.IllegalQueuePathName))
+                try
                 {
-                    return;
+                    queue = CreateQueue(queuePath, identity, settings.UseTransactionalQueues);
                 }
-                if (ex.MessageQueueErrorCode == MessageQueueErrorCode.QueueExists)
+                catch (MessageQueueException ex)
                 {
-                    //Solve the race condition problem when multiple endpoints try to create same queue (e.g. error queue).
-                    return;
+                    if (msmqAddress.IsRemote && (ex.MessageQueueErrorCode == MessageQueueErrorCode.IllegalQueuePathName))
+                    {
+                        return;
+                    }
+                    if (ex.MessageQueueErrorCode == MessageQueueErrorCode.QueueExists)
+                    {
+                        //Solve the race condition problem when multiple endpoints try to create same queue (e.g. error queue).
+                        return;
+                    }
+
+                    Logger.Error($"Could not create queue {msmqAddress} or check its existence. Processing will still continue.", ex);
                 }
-                Logger.Error($"Could not create queue {address} or check its existence. Processing will still continue.", ex);
-            }
-            catch (Exception ex)
-            {
-                Logger.Error($"Could not create queue {address} or check its existence. Processing will still continue.", ex);
             }
 
-            Logger.DebugFormat("Verified that the queue: [{0}] existed", address);
-        }
-
-        static void CreateQueue(string queuePath, string account, bool transactional)
-        {
-            using (var queue = MessageQueue.Create(queuePath, transactional))
+            if (queue == null)
             {
-                SetPermissionsForQueue(queue, account);
+                Logger.Error($"Could not create queue {address} or check its existence. Processing will still continue.");
+                return;
             }
 
-            Logger.DebugFormat("Created queue, path: [{0}], identity: [{1}], transactional: [{2}]", queuePath, account, transactional);
-        }
-
-        static void SetPermissionsForQueue(string queuePath, string account)
-        {
-            using (var messageQueue = new MessageQueue(queuePath))
+            using (queue)
```

Comment:
@bording I didn't want to rely specifically on returning a null, that's why I kept the try/catch block outside. Let me think about it some more...